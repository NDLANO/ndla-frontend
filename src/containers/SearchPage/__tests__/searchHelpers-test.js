import {
  searchResultToLinkProps,
  converSearchStringToObject,
  convertSearchParam,
} from '../searchHelpers';

test('searchHelpers searchResultToLinkProps learningpath', () => {
  const result = {
    id: 500,
    resourceType: 'urn:resourcetype:learningPath',
    path: '/learningpath/500',
    paths: ['/urn:lp:500'],
  };

  expect(searchResultToLinkProps(result)).toMatchSnapshot();
});

test('searchHelpers searchResultToLinkProps article', () => {
  const result = {
    id: 300,
    resourceType: 'urn:resourcetype:subjectMaterial',
    path: '/article/300',
    paths: ['/urn:lp:300'],
  };

  expect(searchResultToLinkProps(result)).toMatchSnapshot();
});

test('searchHelpers searchResultToLinkProps article without paths', () => {
  const result = {
    id: 404,
    resourceType: 'urn:resourcetype:subjectMaterial',
  };

  expect(searchResultToLinkProps(result)).toMatchSnapshot();
});

test('searchHelpers converSearchStringToObject converts search string', () => {
  const locationWithSearch = {
    search:
      '?query=test&page=3&languageFilter=1,2,3&subjects=urn:test:3,urn:test:1,urn:test:2',
  };

  expect(converSearchStringToObject(locationWithSearch)).toMatchSnapshot();
});

test('searchHelpers converSearchStringToObject with no location', () => {
  expect(converSearchStringToObject()).toMatchSnapshot();
});

test('searchHelpers convertSearchParam', () => {
  expect(convertSearchParam()).toEqual(undefined);
  expect(convertSearchParam('NDLA')).toBe('NDLA');
  expect(convertSearchParam(27)).toBe(27);
  expect(convertSearchParam([1, 2, 3, 4, 5, 6, 7, 8, 9, 10])).toBe(
    '1,2,3,4,5,6,7,8,9,10',
  );
});
