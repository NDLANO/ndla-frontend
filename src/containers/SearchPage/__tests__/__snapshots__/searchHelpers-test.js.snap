// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`searchHelpers converSearchStringToObject converts search string 1`] = `
Object {
  "contextFilters": Array [],
  "contextTypes": Array [],
  "languageFilter": Array [
    "1",
    "2",
    "3",
  ],
  "levels": Array [
    "urn:test:3",
    "urn:test:1",
    "urn:test:2",
  ],
  "page": "3",
  "query": "test",
  "relevance": Array [],
  "resourceTypes": Array [],
  "subjects": Array [
    "urn:test:3",
    "urn:test:1",
    "urn:test:2",
  ],
}
`;

exports[`searchHelpers converSearchStringToObject with no location 1`] = `
Object {
  "contextFilters": Array [],
  "contextTypes": Array [],
  "languageFilter": Array [],
  "levels": Array [],
  "relevance": Array [],
  "resourceTypes": Array [],
  "subjects": Array [],
}
`;

exports[`searchHelpers searchResultToLinkProps article 1`] = `
Object {
  "to": "/subjects/article/300",
}
`;

exports[`searchHelpers searchResultToLinkProps article without paths 1`] = `
Object {
  "to": "/404",
}
`;

exports[`searchHelpers searchResultToLinkProps learningpath 1`] = `
Object {
  "to": "/learningpaths/learningpath/500",
}
`;
